sdk/protocol-manager-poc/tests/tests.spec.ts

```diff
@@ -159,6 +174,70 @@ class TokenServiceMock implements ITokenService {
   }
 }
 
+class ContractProviderMock implements IContractProvider {
+  getContractDef<P extends IProtocol['name'], K extends keyof AddressAbiMapsByProtocol[P]>(
+      contractKey: K,
+      protocol: P
+  ): AddressAbiMapsByProtocol[P][K] {
+        const map: AddressAbiMapsByProtocol = {
+            [ProtocolName.Spark]: {
```

Comment:
As mentioned, all addresses are coming from the core-contracts. We should also find a proper way of providing the Abis and have the contract provider there if possible